### PR TITLE
Added AccountClusterTransformer

### PR DESCRIPTION
**Pull request by creukersVF**

### CODE REVIEW

File: vf_portalytics/ml_helpers.py [hunk 1]
```diff
@@ -66,3 +66,31 @@ def transform(self, X, y=None):
...
+
+    def transform(self, X):
+        if 'cluster' not in X.columns:
+            if self.cat_feature == 'vf_category' or self.cat_feature is None:
+                # vf_category means that no category defined
+                X['cluster'] = 0.0
+            elif isinstance(self.cat_feature, (unicode, str)):
+                # category is a feature
+                X['cluster'] = X[self.cat_feature]
+            elif isinstance(self.cat_feature, list):
```
Comment (ctselas7) [inline vf_portalytics/ml_helpers.py:88]: @creukersVF better use one type for `cat_feature` to avoid confusion. You can use always a list and do everything  that is needed

File: vf_portalytics/ml_helpers.py [hunk 2]
```diff
@@ -66,3 +66,31 @@ def transform(self, X, y=None):
...
+        return self
+
+    def transform(self, X):
+        if 'cluster' not in X.columns:
+            if self.cat_feature == 'vf_category' or self.cat_feature is None:
+                # vf_category means that no category defined
+                X['cluster'] = 0.0
+            elif isinstance(self.cat_feature, (unicode, str)):
+                # category is a feature
+                X['cluster'] = X[self.cat_feature]
```
Comment (ctselas7) [inline vf_portalytics/ml_helpers.py:87]: what if `self.cat_feature` is not a column in X?

File: vf_portalytics/ml_helpers.py [hunk 3]
```diff
@@ -66,3 +66,31 @@ def transform(self, X, y=None):
...
+    """A custom transformer that could use a list of accounts to create clusters"""
+
+    def __init__(self, cat_feature=None):
+        self.cat_feature = cat_feature
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X):
+        if 'cluster' not in X.columns:
```
Comment (ctselas7) [inline vf_portalytics/ml_helpers.py:81]: Better 
```
if 'cluster' in X.columns:
   return X
```
Its more clear of what is being done afterwards

File: vf_portalytics/ml_helpers.py [hunk 4]
```diff
@@ -66,3 +66,31 @@ def transform(self, X, y=None):
             return self.transformer.transform(X, y)
         else:
             return self.transformer.transform(X, y.iloc[:, 0])
+
+
+class AccountClusterTransformer(BaseEstimator, TransformerMixin):
+    """A custom transformer that could use a list of accounts to create clusters"""
+
+    def __init__(self, cat_feature=None):
```
Comment (ctselas7) [inline vf_portalytics/ml_helpers.py:74]: Could you add some info of what the cat_feature is?

File: tests/test_account_cluster_transformer.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    train_x = pd.DataFrame(train_x)
+    train_y = pd.DataFrame(train_y)
+
+    cat_feature = ['A', 'B', 'C', 'D']
+    transformer = AccountClusterTransformer(cat_feature)
+    transformed_x = transformer.fit_transform(train_x, train_y)
+
+    assert all(train_x['Var_1'] == transformed_x['Var_1'])
+
+    assert ('cluster' in transformed_x.columns)
```
Comment (ctselas7) [inline tests/test_account_cluster_transformer.py:24]: If you use `cat_feature = ['A', 'B', 'C', 'D']` the `AccountClusterTransformer` will just create a new column named `cluster` equal to  `account_banner`. This doesn't check edge cases. I would propose you make a smaller dataframe and test how the new code behaves in different cases

**Review by ctselas7 — CHANGES_REQUESTED** [review]

File: vf_portalytics/ml_helpers.py [hunk 6]
```diff
@@ -66,3 +70,42 @@ def transform(self, X, y=None):
...
+class AccountClusterTransformer(BaseEstimator, TransformerMixin):
+
+    """
+
+    A custom transformer that could use a list of accounts to create clusters
+    """
+
+    def __init__(self, cat_feature_list=None):
+        """
+        :param cat_feature_list: List of accounts in account_banner feature
```
Comment (ctselas7) [inline vf_portalytics/ml_helpers.py:84]: Its not always an account in account_banner

File: vf_portalytics/ml_helpers.py (hunk 3 above)
Comment (ctselas7) [reply vf_portalytics/ml_helpers.py:92]: @creukersVF this?

**Review by ctselas7 — APPROVED** [review]
